tables/AlternateMans/UvwFile.h

```diff
@@ -0,0 +1,403 @@
+          "columns: the measurement set may be damaged");
+    }
+    rows_per_block_ = reinterpret_cast<uint64_t&>(data[8]);
+    reference_antenna_ = reinterpret_cast<uint64_t&>(data[16]);
+    n_antenna_ = reinterpret_cast<uint64_t&>(data[24]);
+  }
+
+  void WriteHeader() {
+    unsigned char data[kHeaderSize];
+    std::copy_n(kMagicHeaderTag, 8, data);
+    reinterpret_cast<uint64_t&>(data[8]) = rows_per_block_;
+    reinterpret_cast<uint64_t&>(data[16]) = reference_antenna_;
+    reinterpret_cast<uint64_t&>(data[24]) = n_antenna_;
+    file_.WriteHeader(data);
+  }
+
+  bool IsSet(size_t antenna) const {
+    return block_uvws_.size() > antenna &&
+           block_uvws_[antenna] != kUnsetPosition;
+  }
+  static bool AreNear(std::array<double, 3> a, std::array<double, 3> b) {
+    return AreNear(a[0], b[0]) && AreNear(a[1], b[1]) && AreNear(a[2], b[2]);
+  }
+  static bool AreNear(double a, double b) {
+    const double magnitude = std::max({1e-5, std::fabs(a), std::fabs(b)});
+    return (std::fabs(a - b) / magnitude) < 1e-5;
+  }
+  static std::string UvwAsString(const std::array<double, 3>& uvw) {
+    std::ostringstream str;
+    str << "[" << uvw[0] << ", " << uvw[1] << ", " << uvw[2] << "]";
+    return str.str();
+  }
+
+  /**
+   * The header:
+   * char[8] "Uvw-col\0" (=kMagicHeaderTag)
+   * uint64_t rows_per_block
+   * uint64_t reference_antenna
+   * uint64_t n_antenna
+   */
+  constexpr static size_t kHeaderSize = 32;
+  constexpr static const char kMagicHeaderTag[8] = "Uvw-col";
+  constexpr static std::array<double, 3> kUnsetPosition = {
+      std::numeric_limits<double>::max(), std::numeric_limits<double>::max(),
+      std::numeric_limits<double>::max()};
+  BufferedColumnarFile file_;
+  /**
+   * Number of rows in the Uvw column. It is increased when writing a new
+   * Uvw row using @ref WriteUvw(). This concept of a "row" is different
+   * from a row in the BufferedColumnarFile (i.e., @c file_ ), as for each
+   * block (see below), only the uvw per antenna are stored, and these form
+   * the rows in that file.
+   */
+  uint64_t n_rows_ = 0;
+  /**
+   * A "block" is a contiguous number of baselines that together form
+   * one timestep. This is the same as saying they form one (triangular)
+   * correlation matrix. It can have auto-correlation but this is not
+   * required. A block may have missing antennas, but has some constraints
+   * on the ordering; see the class description. Per block, the UVW values
+   * are calculated per antenna. This value specifies the number of table
+   * rows in one block, i.e. the number of set elements in the correlation
+   * matrix.
+   */
+  uint64_t rows_per_block_ = 0;
+  uint64_t active_block_ = 0;
+  size_t reference_antenna_ = 0;
+  // This value is used to determine the first baseline in the data, which is
+  // the baseline (reference_antenna_, start_antenna_2_).
+  size_t start_antenna_2_ = 0;
+  size_t n_antenna_ = 0;
```

Comment:
Although `n_antenna_` is a clear name, `n_antennas_` would be more consistent with `n_rows_`.
```suggestion
  size_t n_antennas_ = 0;
```